nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -12,24 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
 from typing import Dict, Optional
 
+import hydra
 import torch
+from omegaconf import MISSING, DictConfig
 from torch.utils.data import DataLoader
 
 from nemo.collections.common.losses import CrossEntropyLoss
 from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
 from nemo.collections.nlp.data.text_classification import TextClassificationDataDesc, TextClassificationDataset
 from nemo.collections.nlp.modules.common import SequenceClassifier
 from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
-from nemo.core.classes import typecheck
-from nemo.core.classes.modelPT import ModelPT
+from nemo.core.classes.common import typecheck
+from nemo.core.classes.modelPT import ModelPT, ModelPTConfig
 from nemo.core.neural_types import NeuralType
 from nemo.utils.decorators import experimental
 
 __all__ = ['TextClassificationModel']
 
 
+@dataclass
+class TextClassificationDataConfig:
+    prefix: str = None
+    batch_size: int = 32
+    shuffle: bool = False
+    num_samples: int = -1
+    num_workers: int = 2
+    pin_memory: bool = False
+    use_cache: bool = True
+    drop_last: bool = True
+
+
+@dataclass
+class LanguageModelConfig:
+    pretrained_model_name: Optional[str] = 'roberta-base'
+    bert_config: Optional[DictConfig] = None
+    max_seq_length: int = 36
+
+
+@dataclass
+class TextClassificationHeadConfig:
+    num_output_layers: int = 2
+    fc_dropout: float = 0.1
+
+
+@dataclass
+class TextClassificationModelConfig(ModelPTConfig):
+    language_model: LanguageModelConfig = MISSING
```

Comment:
why do these have `@dataclass` decorator? Can we have just a single Model config?

Reply:
@dataclass decorator is necessary for config objects. But, yes, let's use Config objects for Models only.

Reply:
with dataclass we can have structured configs and inheritance. Hopefully creating a common look and feel across the collection and across nemo.